client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,237 @@
+        cls.log.info(create_context.get_current_task_entity())
+        if not cls.instance_matches_plugin_families(instance):
+            return []
+
+        # will be reworked when CreateContext contains settings and task types
+        project_name = create_context.project_name
+        project_settings = (
+            create_context.get_current_project_settings()
+        )
+
+        profiles = (
+            project_settings["deadline"]["publish"][cls.__name__]["profiles"])
+
+        if not profiles:
+            return []
+
+        host_name = create_context.host_name
+
+        task_name = instance["task"]
+        folder_path = instance["folderPath"]
+        task_entity = create_context.get_task_entity(folder_path, task_name)
+
+        profile = filter_profiles(
+            profiles,
+            {
+                "host_names": host_name,
+                "task_types": task_entity["taskType"],
+                "task_names": task_name,
+                # "product_type": product_type
+            }
+        )
+        overrides = set(profile["overrides"])
+        if not profile or not overrides:
+            return []
+
+        defs = [
+            UISeparatorDef("deadline_defs_starts"),
+        ]
+
+        defs.extend(cls._get_artist_overrides(overrides, profile))
+
+        # explicit
+        defs.append(
+            TextDef(
+                "frames",
+                label="Frames",
+                default="",
+                tooltip="Explicit frames to be rendered. (1, 3-4)"
+            )
+        )
+
+        defs.append(
+            UISeparatorDef("deadline_defs_end")
+        )
+
+        return defs
+
+    @classmethod
+    def _get_artist_overrides(cls, overrides, profile):
+        """Provide list of Defs that could be filled by artist"""
+        # should be matching to extract_jobinfo_overrides_enum
+        override_defs = [
+            NumberDef(
+                "chunkSize",
+                label="Frames Per Task",
+                default=1,
+                decimals=0,
+                minimum=1,
+                maximum=1000
+            ),
+            NumberDef(
+                "priority",
+                label="Priority",
+                decimals=0
+            ),
+            TextDef(
+                "department",
+                label="Department",
+                default="",
+            ),
+            TextDef(
+                "limit_groups",
+                label="Limit Groups",
+                # multiline=True,  TODO - some DCC might have issues with storing multi lines
+                default="",
+                placeholder="machine1,machine2"
+            ),
+            TextDef(
+                "job_delay",
+                label="Delay job (timecode dd:hh:mm:ss)",
+                default=""
+            )
+        ]
+        defs = []
+        # The Arguments that can be modified by the Publisher
+        for attr_def in override_defs:
```

Comment:
I would avoid all of this, and do it explicitly only on keys that are list in settings. So you don't create list of attribute definitions, and afterwards change their default and create new list of attribute definitions.

Reply:
The list of attr defs could be dynamic, studios might want to show/override different attributes per DCC or task type.

Reply:
We should really start making the Limit Groups a multiselection enum with the actual groups retrieved/cached from the server. That'd at least simplify how we're passing those values on to the instance - however it may mean more refactoring on the creator side to ensure we query that data from the server (and have that dynamic based on 'which server' because technically we support multiple deadline servers)

It'd basically match this: https://community.ynput.io/t/deadline-pools-and-groups-dropdown-menu/1770/2?u=bigroy

Reply:
Ok, changed